components/object_store/src/disk_cache.rs

```diff
@@ -686,6 +712,67 @@ mod test {
         assert!(test_file_exists(&store.cache_dir, &location, &(48..64)));
     }
 
+    #[tokio::test]
+    async fn test_disk_cache_remove_cache_file_two_partition() {
+        let page_size = 16;
+        // 51 byte
+        let data = b"a b c d e f g h i j k l m n o p q r s t u v w x y z";
+        let location = Path::from("remove_cache_file_two_partition.sst");
+        // partition_cap: 64 / 16 / 2 = 2
+        let store = prepare_store(page_size, 64, 1).await;
+        let mut buf = BytesMut::with_capacity(data.len() * 8);
+        // extend 8 times
+        for _ in 0..8 {
+            buf.extend_from_slice(data);
+        }
+        store.inner.put(&location, buf.freeze()).await.unwrap();
+        // 0..16: partition 1
+        // 16..32 partition 0
+        // 32..48 partition 0
+        // 48..64 partition 0
+        // 64..80 partition 0
```

Comment:
If total capacity is 64, then range large than it will be trimmed to 64. You should increase capacity to make this consumption right.